### ISSUE TITLE
SIM card contacts handling

### ISSUE THREAD

**Issue by qwertyfinger** [SimpleMobileTools/Simple-Contacts#707]
This is created as a discussion about the best way to handle SIM card contacts.
I believe it's related to most, if not all, of these issues: #181, #702, #663, #675, #677.

I think the only way to reliably display, edit, and delete SIM card contacts across all devices is by using SIM card storage that's available via `content://icc/adn` URI.

When using a general contacts database, the behavior differs from device to device. Some manufacturers choose to include SIM contacts in it in some way, but on other devices (including pure Android on Pixel devices), they are not included in the general database at all.

But the SIM card `content://icc/adn` database can only contain the name and phone number fields, so we can't store any other data for these contacts. It's natural since when moving SIM cards between devices, all the extra information is never migrated.

- One option is to disable editing all the other fields when the contact is stored on a SIM card. A user can export SIM card contact to another storage, and then all the fields will be editable. When moving contact from another storage to a SIM card – only the name and phone would be migrated.
- Another, more complicated, is to automatically create a synced copy of a SIM contact in the local storage that will contain all the extra info, but only name and phone number will be synced with SIM card storage. This way user will have all the info while he's using the current device, and if he moves the SIM card between devices, only the name and number will be migrated.

**Comment by tibbi** [comment]
let's not complicate it, the vast majority of people uses the name and number only. If that will work reliably on SIM cards, I will be happy.

**Comment by qwertyfinger** [comment]
Yes, I like the simpler approach more too. Let's try to do it this way then.

**Comment by inolil** [comment]
Hi,
Is there a workaround for this problem ? It's quite a big issue not to be able to save contacts numbers on a phone.
A precision regarding my own situation : I don't like google, so I disabled all google applications et services. Including Google Play Services.
Everything works fine, thanks to Simple Mobile Tools, except the contacts : all changes are lost after reboot...
Regards
